File: GuardeSoftwareAPI/Dtos/Client/GetClientsRequest.cs
namespace GuardeSoftwareAPI.Dtos.Client
{
    public class GetClientsRequestDto
    {
        public int PageNumber { get; set; } = 1;
        public int PageSize { get; set; } = 10;
        public string SortField { get; set; } = "FullName";
        public string SortDirection { get; set; } = "asc";
        public string? SearchTerm { get; set; }
        public string? StatusFilter { get; set; }
        public bool? Active { get; set; } = true;
        public int? WarehouseId { get; set; }
        public List<int>? WarehouseIds { get; set; }
        public List<int>? ExcludedWarehouseIds { get; set; }
        public string? AdvancedFilter { get; set; }
        public List<string>? AdvancedFilters { get; set; }
        public List<string>? ExcludedAdvancedFilters { get; set; }
        public List<string>? IvaConditions { get; set; }
        public List<string>? ExcludedIvaConditions { get; set; }
        public List<int>? BillingTypeIds { get; set; }
        public List<int>? ExcludedBillingTypeIds { get; set; }
        public List<int>? PreferredPaymentMethodIds { get; set; }
        public List<int>? ExcludedPreferredPaymentMethodIds { get; set; }
        public List<int>? LockerTypeIds { get; set; }
        public List<int>? ExcludedLockerTypeIds { get; set; }
        public List<int>? PaymentDays { get; set; }
        public List<int>? ExcludedPaymentDays { get; set; }
    }
}

File: GuardeSoftwareClient/src/app/core/dtos/client/GetClientsRequest.ts
export interface GetClientsRequest {
  pageNumber?: number;
  pageSize?: number;
  sortField?: string;
  sortDirection?: 'asc' | 'desc';
  searchTerm?: string;
  statusFilter?: string;
  active?: boolean;
  warehouseId?: number;
  warehouseIds?: number[];
  excludedWarehouseIds?: number[];
  advancedFilter?: string;
  advancedFilters?: string[];
  excludedAdvancedFilters?: string[];
  ivaConditions?: string[];
  excludedIvaConditions?: string[];
  billingTypeIds?: number[];
  excludedBillingTypeIds?: number[];
  preferredPaymentMethodIds?: number[];
  excludedPreferredPaymentMethodIds?: number[];
  lockerTypeIds?: number[];
  excludedLockerTypeIds?: number[];
  paymentDays?: number[];
  excludedPaymentDays?: number[];
}
